### PR TITLE
Add `haddock3` support

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR adds support to `haddock3`.

The overall functionallity is the same, the major difference is that the input `.yaml` file is slightly different to account for haddock3's modules. See the example below.

Same functionalities extent to `haddock3`; parameters defined in the `yaml` file are checked against the module's `defaults.yaml` file (except mandatory and optional parameters). The selection of the restraints and the receptor also follow pattern matching.

```yaml
general:
  # ...
  # same as haddock2.4
  # ...

scenarios:
  - name: true-interface

    parameters:
      general:
        # these are the general parameters defined in haddock3's
        #  `run.toml` header - there's currently limited support for this
        mode: local
        ncores: 1

      modules:
        # these are the modules to be executed in the order they are defined
        order: [topoaa, rigidbody, seletop, flexref, emref]
        # these are the parameters for each module
        topoaa:
          autohis: true

        rigidbody:
          sampling: 10
          ambig_fname: "_ti.tbl"

        seletop:
          select: 2

        # if no custom parameters are being used, it can be empty
        flexref:

        emref:

  - name: center-of-mass
    parameters:

      general:
        mode: local
        ncores: 1

      modules:
        order: [topoaa, rigidbody]

        topoaa:
          autohis: true

        rigidbody:
          sampling: 2
          cmrest: true
```

### CODE REVIEW

**Comment by rvhonorato** [comment]
@amjjbonvin @mgiulini you can try to run the example from the repository or check a working example here: `/trinity/login/rodrigo/projects/benchmark-tools` - we can discuss this IRL and I'll update the PR later (:

File: haddock3.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+#===============================================================================
+
+HADDOCK3_DIR="$HOME/repos/haddock3"
+
+# shellcheck source=/dev/null
+source "$HADDOCK3_DIR/venv/bin/activate" || exit
+
```
Comment (mgiulini) [inline haddock3.sh:8]: here I would add the commands for activating haddock3 using conda, something like the following

```suggestion
# if your haddock3 installation uses conda
# conda init
# source $HOME/anaconda3/bin/activate
# source activate haddock3
```

**Review by mgiulini — APPROVED** [review]
nice addition! I ran the example benchmark with my version of haddock3, everything went quite smoothly

File: haddock3.sh (hunk 1 above)
Comment (rvhonorato) [reply haddock3.sh:8]: Indeed, thanks!

File: haddock3.sh [hunk 2]
```diff
@@ -5,7 +5,10 @@ HADDOCK3_DIR="$HOME/repos/haddock3"
 
 # shellcheck source=/dev/null
 source "$HADDOCK3_DIR/venv/bin/activate" || exit
-
+# if your haddock3 installation uses conda
```
Comment (amjjbonvin) [inline haddock3.sh:8]: For conda use I would rather define:

```
# if your haddock3  installation uses conda
##source $HOME/miniconda3/etc/profile.d/conda.sh
##conda activate haddock3
```

The `bin/activate` command if I am correct add the conda initialisation to your .cshrc / .profile files.

Comment (amjjbonvin) [reply haddock3.sh:8]: This is what I used for the workshop on the Fugaku supercomputer (and for benchmarking on Snellius)

**Review by amjjbonvin — COMMENTED** [review]
For Conda I would rather define:

**Comment by rvhonorato** [comment]
I've written a tutorial on how to setup/configure/run the code with both HADDOCK2.4 and HADDOCK3.0, its now based on this PR -> https://github.com/haddocking/haddocking.github.io/pull/625

**Pull request merged**